File: calculator/src/test/java/com/grijesh/calculator/test/calculator/CalculatorTest.java
package com.grijesh.calculator.test.calculator;

import com.grijesh.calculator.Calculator;
import cucumber.api.java.en.Given;
import cucumber.api.java.en.Then;
import cucumber.api.java.en.When;
import org.junit.Assert;

/**
 * Created by grijesh on 17/10/15.
 */
public class CalculatorTest {

    private Calculator calculator;

    @Given("^Calculator is turned on$")
    public void calculator_is_turned_on() throws Throwable {
       calculator = new Calculator();
    }

    @Given("^reset calculator$")
    public void resetCurrentState() throws Throwable{
        calculator.setCurrentState(0);
    }

    @When("^I add (.+)$")
    public void addFirst(int firstNumber) throws Throwable {
        calculator.add(firstNumber);
    }

    @When("^add (.+)$")
    public void addSecond(int secondNumber) throws Throwable {
        calculator.add(secondNumber);
    }

    @Then("^result should be (.+)$")
    public void result_should_be(int result) throws Throwable {
        Assert.assertTrue(calculator.getCurrentState()== result);
    }

    @When("^subtract (.+)$")
    //Replaced \\d+ to .+ to support double or float number
    public void subtract(int number) throws Throwable {
        calculator.subtract(number);
    }

}

File: scala-calculator/src/test/scala/com/grijesh/calculator/test/CalculatorTest.scala
package com.grijesh.calculator.test

import com.grijesh.calculator.Calculator
import cucumber.api.scala.{ScalaDsl, EN}
import org.junit.Assert._

/**
  * Created by grijesh on 13/3/16.
  */
class CalculatorTest extends ScalaDsl with EN{

  private var calc:Calculator = null

  Given("""^Calculator is turned on$"""){ () =>
    calc = new Calculator
  }
  When("""^I add (\d+)$"""){ (a:Int) =>
    calc add a
  }
  When("""^add (\d+)$"""){ (b:Int) =>
    calc add b
  }
  Then("""^result should be (\d+)$"""){ (result:Int) =>
    assertEquals(result, calc result)
  }
  When("""^subtract (\d+)$"""){ (a:Int) =>
    calc subtract a
  }
  When("""^add (\d+).(\d+)$"""){ (a:Int, b:Int) =>
    //// Express the Regexp above with the code you wish you had
  }
  Then("""^result should be (\d+).(\d+)$"""){ (a:Int, b:Int) =>
    //// Express the Regexp above with the code you wish you had
  }

}
